woodwork/utils.py

```diff
@@ -298,3 +299,149 @@ def _parse_logical_type(logical_type, name):
         return logical_type
     else:
         raise TypeError(f"Invalid logical type specified for '{name}'")
+
+
+def concat(objs, axis=1, join='outer', validate_schema=True):
+    """
+    Concatenate Woodwork objects along a particular axis. Cannot concatenate DataFrames
+    with conflicting Woodwork index or time index columns. If DataFrames have the same Woodwork
+    index or time index columns, the first appearance of the column will be included in the
+    concatenated DataFrame. As Woodwork does not allow duplicate column names,
+    will not allow duplicate columns at concatenation.
+
+    Args:
+        objs (list[Series, DataFrame]): The Woodwork objects to be concatenated. If Woodwork
+            is not initialized on any of the objects, type inference will be performed.
+        axis ({0/’index’, 1/’columns’}, optional): The axis to concatenate along. Defaults to 1.
+        join ({'inner', 'outer'}, optiona): How to handle indexes on other axis. Defaults to 'outer'.
+        validate_schema (bool, optional): Whether validation should be performed on the typing information
+            for the concatenated DataFrame. Defaults to True.
+
+    Returns:
+        object, type of objs: When objs contains at least one DataFrame, a Woodwork DataFrame is returned.
+            When concatenating along the columns (axis=1), a Woodwork DataFrame is returned.
+            When concatenating along the index (axis=1) with only series, a series is returned
+            without Woodwork initialized.
+    """
+    if not objs:
+        raise ValueError('No objects to concatenate')
+
+    # --> consider adding sort
+    # --> consider adding ignore index
+    table_name = ''
+    logical_types = {}
+    semantic_tags = {}
+    col_descriptions = {}
+    col_metadata = {}
+    table_metadata = {}
+    use_standard_tags = {}
+
+    index = None
+    time_index = None
+
+    concat_objs = []
+    # Record the typing information for all the columns that have Woodwork schemas
+    col_names_seen = set()
+    for obj in objs:
+        ww_columns = {}
+        obj_to_add = obj
+        if obj is None:
+            # --> not confident that this is the right choice. We need to be able to use the elements of objs to get the library type so it's hard if any element can be None
+            raise ValueError('Cannont include None in list of objects to concatenate')
```

Comment:
Only Pandas allows `None` as inputs to concat. It just ignores it. But Because we rely on checking the type of an object in `objs` to know the library to use for concat, having possible `None` values in `objs` means we'll need a more complex check to determine the library, and since two of them don't allow None as input, I figured we could just disallow it.

Reply:
I'm not sure we should add this check at all. The code for checking the library defaults to pandas, and it's probably rare that someone would pass weird values `None` or `3` instead of a series or dataframe.

Reply:
It won't get to the concat call, because the check for if `isinstance(obj.ww.schema, ww.table_schema.TableSchema)` will raise some sort of `NoneType has no attribute ww` error. Then again, if we're not supporting None, maybe that's a clear enough error.

Reply:
Removed the specific None check for simplicity